src/trades.py

```diff
@@ -0,0 +1,74 @@
+"""
+Definition of trades, orders, and executions.
+"""
+
+from fractions import Fraction
+
+
+class Trade:
+    """
+    Class for executed orders.
+    """
+
+    def __init__(self, data, execution):
+        self.data = data
+        self.execution = execution
+
+    def adapt_execution_to_gas_price(self, gas_price, gas_price_adapted):
+        """
+        Given an executed order which was executed ata time with gas price `gas_price`, adapt the
+        execution to what it would have been with gas price `gas_price_adapted`.
+        """
+        self.execution.adapt_execution_to_gas_price(gas_price, gas_price_adapted)
+
+
+class OrderData:
+    """
+    Class for order data.
+    """
+
+    def __init__(
+        self,
+        limit_buy_amount,
+        limit_sell_amount,
+        precomputed_fee_amount,
+        buy_token,
+        sell_token,
+        is_sell_order,
+        is_partially_fillable,
+    ):
+        self.limit_buy_amount = limit_buy_amount
+        self.limit_sell_amount = limit_sell_amount
+        self.precomputed_fee_amount = precomputed_fee_amount
+        self.buy_token = buy_token
+        self.sell_token = sell_token
+        self.is_sell_order = is_sell_order
+        self.is_partially_fillable = is_partially_fillable
+
+
+class OrderExecution:
```

Comment:
Not a single one of these functions in this file have type declarations! This means I could do the following:


```py
my_evil_execution_instance = OrderExecution("Anything", "I", "want")

my_evil_execution_instance.adapt_execution_to_gas_price("Schfifty Five", "Schwenty One")
```

I wonder what that will do? Do we really want to test things like that?

Reply:
The move to data classes should automatically to the type checking now, right?